core/runtime/src/main/java/io/quarkus/runtime/configuration/MemorySizeConverter.java

```diff
@@ -0,0 +1,105 @@
+package io.quarkus.runtime.configuration;
+
+import static io.quarkus.runtime.configuration.ConverterSupport.DEFAULT_QUARKUS_CONVERTER_PRIORITY;
+
+import java.math.BigInteger;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import javax.annotation.Priority;
+
+import org.eclipse.microprofile.config.spi.Converter;
+
+/**
+ * A converter to support data sizes.
+ */
+@Priority(DEFAULT_QUARKUS_CONVERTER_PRIORITY)
+public class MemorySizeConverter implements Converter<MemorySize> {
+    private static final String EMPTY = "";
+
+    private static final String DIGITS = "\\d+";
+    private static final String B = "[Bb]";
+    private static final String KB = "[Kk]";
+    private static final String MB = "[Mm]";
+    private static final String GB = "[Gg]";
+    private static final String TB = "[Tt]";
+    private static final String PB = "[Pp]";
+    private static final String EB = "[Ee]";
+    private static final String ZB = "[Zz]";
+    private static final String YB = "[Yy]";
+
+    private static final String BEGIN = "^";
+    private static final String END = "$";
+    private static final Predicate<String> isNumeric = Pattern.compile(BEGIN + DIGITS + END).asPredicate();
+    private static final Predicate<String> isInByteScale = Pattern.compile(BEGIN + DIGITS + B + END).asPredicate();
+    private static final Predicate<String> isInKiloByteScale = Pattern.compile(BEGIN + DIGITS + KB + END).asPredicate();
+    private static final Predicate<String> isInMegaByteScale = Pattern.compile(BEGIN + DIGITS + MB + END).asPredicate();
+    private static final Predicate<String> isInGigaByteScale = Pattern.compile(BEGIN + DIGITS + GB + END).asPredicate();
+    private static final Predicate<String> isInTeraByteScale = Pattern.compile(BEGIN + DIGITS + TB + END).asPredicate();
+    private static final Predicate<String> isInPetaByteScale = Pattern.compile(BEGIN + DIGITS + PB + END).asPredicate();
+    private static final Predicate<String> isInExaByteScale = Pattern.compile(BEGIN + DIGITS + EB + END).asPredicate();
+    private static final Predicate<String> isInZettaByteScale = Pattern.compile(BEGIN + DIGITS + ZB + END).asPredicate();
+    private static final Predicate<String> isInYottaByteScale = Pattern.compile(BEGIN + DIGITS + YB + END).asPredicate();
```

Comment:
I'm wondering whether we should be initializing these lazily/on-demand. I guess these aren't necessarily going to be used on every build by every app.

Reply:
There's really no reason to do it this way. Better to have a single pattern which matches the number and optional suffix.  Then just test the matched suffix to see what multiplier to apply.  It should be one pattern instance and one matcher instance using capture groups to isolate the number and suffix parts.

Reply:
@dmlloyd indeed,  I'll address this as proposed. This was my second option. I opted for the current solution as it seemed obvious.

Reply:
@aloubyansky @dmlloyd Done. It is ready for another round of reviews.